core/src/service/file_service.rs

```diff
@@ -575,17 +578,16 @@ impl Tx<'_> {
         let base_metadata = self.get_all_metadata(RepoSource::Base)?;
         let local_metadata = self.get_all_metadata(RepoSource::Local)?;
         self.insert_metadata_given_decrypted_metadata(
-            config,
             RepoSource::Base,
             &base_metadata,
             base_metadata_changes,
         )?;
         self.insert_metadata_given_decrypted_metadata(
-            config,
             RepoSource::Local,
             &local_metadata,
             local_metadata_changes,
-        )
+        )?;
+        self.insert_new_docs(config, &local_metadata, local_metadata_changes)
```

Comment:
`insert_metadata_given_decrypted_metadata` can't be responsible for adding new documents because to tell if a document is new, you need to read the current state, but in the event of cycle resolution the file tree will have a cycle in-between the two calls to `insert_metadata_given_decrypted_metadata` here. Also of note, `insert_new_docs` must be called after `insert_metadata_given_decrypted_metadata` so that it can check that the metadata for the docs it is inserting already exists, but it must use the version of `local_metadata` from before any calls to `insert_metadata_given_decrypted_metadata` because it only knows which documents are new based on which documents are missing from the local metadata (and they aren't missing anymore after the metadata changes are applied).